src/in-depth/signals.md

```diff
@@ -1 +1,125 @@
 # Signal handling
+
+Processes
+like command line applications
+need to react to signals sent by the operating system.
+The most common example is probably <kbd>Ctrl</kbd>+<kbd>C</kbd>,
+the signal that typically tells a process to terminate.
+To handle signals in Rust programs
+you need to consider how you can receive these signals
+as well as how you can react to them.
+
+<aside>
+
+**Note:**
+If your applications does not require special cleanup,
+the default handling is fine
+(i.e. exit immediately and let the OS cleanup).
+In that case:
+No need to do what this chapter tells you!
+For applications that need to clean up after themselves,
+for example to correct close network connections,
+remove temporary files,
+or reset system settings,
+this chapter is however very relevant!
+
+</aside>
+
+## Differences between operating systems
+
+On Unix systems
+(like Linux, macOS, and FreeBSD)
+a process can receive [signals]
+It can either react to them
+in a default (OS-provided) way,
+catch the signal and handle them in a program-defined way,
+or ignore the signal entirely.
+
+[signals]: https://manpages.ubuntu.com/manpages/bionic/en/man7/signal.7.html
+[signal-safety man page]: http://manpages.ubuntu.com/manpages/bionic/man7/signal-safety.7.html
```

Comment:
This link is unused?

Reply:
Yeah, I meant to link to it but didn't find a good place, as it's a non-problem if you use a crate calls you hook outside of the signal handler.